src/orion/core/worker/experiment.py

```diff
@@ -215,71 +230,34 @@ def retrieve_result(self, trial, *args, **kwargs):
         """See :func:`~orion.storage.BaseStorageProtocol.retrieve_result`"""
         return self._storage.retrieve_result(trial, *args, **kwargs)
 
-    def update_trial(self, *args, **kwargs):
-        """See :func:`~orion.storage.BaseStorageProtocol.update_trial`"""
-        return self._storage.update_trial(*args, **kwargs)
+    def set_trial_status(self, *args, **kwargs):
+        """See :func:`~orion.storage.BaseStorageProtocol.set_trial_status`"""
+        return self._storage.set_trial_status(*args, **kwargs)
 
-    def reserve_trial(self, score_handle=None, _depth=1):
+    def reserve_trial(self, score_handle=None):
         """Find *new* trials that exist currently in database and select one of
         them based on the highest score return from `score_handle` callable.
 
-        :param score_handle: A way to decide which trial out of the *new* ones to
-           to pick as *reserved*, defaults to a random choice.
-        :type score_handle: callable
-        :param _depth: recursion depth only used for logging purposes can be ignored
-        :return: selected `Trial` object, None if could not find any.
-        """
-        log.debug('%s reserving trial with (score: %s)', '<' * _depth, score_handle)
-        if score_handle is not None and not callable(score_handle):
-            raise ValueError("Argument `score_handle` must be callable with a `Trial`.")
-
-        self.fix_lost_trials()
-
-        query = dict(
-            experiment=self._id,
-            status={'$in': ['new', 'suspended', 'interrupted']}
-            )
-
-        new_trials = self.fetch_trials(query)
-        log.debug('%s Fetched (trials: %s)', '<' * _depth, len(new_trials))
-
-        if not new_trials:
-            log.debug('%s no new trials found', '<' * _depth)
-            return None
-
-        if score_handle is not None and self.space:
-            scores = list(map(score_handle,
-                              map(lambda x: trial_to_tuple(x, self.space), new_trials)))
-            scored_trials = zip(scores, new_trials)
-            best_trials = filter(lambda st: st[0] == max(scores), scored_trials)
-            new_trials = list(zip(*best_trials))[1]
-        elif score_handle is not None:
-            log.warning("While reserving trial: `score_handle` was provided, but "
-                        "parameter space has not been defined yet.")
-
-        selected_trial = random.sample(new_trials, 1)[0]
-        log.debug('%s selected (trial: %s)', '<' * _depth, selected_trial)
-
-        update = dict(status='reserved', heartbeat=datetime.datetime.utcnow())
+        Parameters
+        ----------
+        score_handle: Optional[Callable[[Trial], int]]
```

Comment:
Deprecation should be mentioned in the docstring. And type should be `callable object, optional` ([ref]( https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard))